src/apps/ocioperf/main.cpp

```diff
@@ -573,101 +459,197 @@ int main(int argc, const char **argv)
                 m.resume();
                 cpuProcessor = optProcessor->getOptimizedCPUProcessor(inBitDepth,
                                                                       outBitDepth,
-                                                                      OCIO::OPTIMIZATION_DEFAULT);
+                                                                      optimFlags);
                 m.pause();
             }
         }
 
         std::cout << std::endl << std::endl;
         std::cout << "Image processing statistics:" << std::endl << std::endl;
 
+        // Create an arbitrary 4K RGBA image.
 
-        if(testType==0 || testType==-1)
+        static constexpr size_t width  = 3840;
+        static constexpr size_t height = 2160;
+        static constexpr size_t numChannels = 4;
+
+        static constexpr size_t maxElts = width * height;
+
+        std::vector<float> img_f32_ref;
+        std::vector<uint16_t> img_ui16_ref;
+
+        // Generate a synthetic image by emulating a LUT3D identity algorithm that steps through
+        // many different colors.  Need to avoid a constant image, simple gradients, or anything
+        // that would result in more cache hits than a typical image.  Also, want to step through a 
+        // wide range of colors, including outside [0,1], in case some algorithms are faster or
+        // slower for certain colors.
+
+        static constexpr size_t length   = 64;
```

Comment:
[Question] How was this size picked? 64^3 is way lower than 3840x2160 so I think you have lot of repetition in the image but that might be by design?

Reply:
Good point Remi, I would suggest increasing it to 201.